marlowe-playground-client/src/SimulationPage/View.purs

```diff
@@ -0,0 +1,564 @@
+
+    expirationSlot = state ^. (_marloweState <<< _Head <<< _contract <<< to contractMaxTime)
+
+    contractMaxTime = case _ of
+      Nothing -> "Closed"
+      Just contract ->
+        let
+          t = (_.maxTime <<< unwrap <<< timeouts) contract
+        in
+          if t == zero then "Closed" else show t
+
+    indicator name value =
+      div_
+        [ span
+            [ class_ bold ]
+            [ text $ name <> ": " ]
+        , span_ [ text value ]
+        ]
+  in
+    div
+      [ classes [ flex, justifyBetween ] ]
+      [ indicator "current slot" currentSlot
+      , indicator "expiration slot" expirationSlot
+      ]
+
+------------------------------------------------------------
+actionWidget ::
+  forall p.
+  State ->
+  HTML p Action
+actionWidget state =
+  cardWidget "Actions"
+    $ div [ classes [] ]
+        [ ul [ class_ (ClassName "participants") ]
+            if (Map.isEmpty possibleActions) then
+              [ text "No valid inputs can be added to the transaction" ]
+            else
+              (actionsForParties possibleActions)
+        , div [ classes [ ClassName "transaction-btns", flex, justifyCenter ] ]
+            [ button
+                [ classes [ btn, bold, (Classes.disabled $ not isEnabled), spaceRight ]
+                , disabled $ not $ hasHistory state
+                , onClick $ const $ Just Undo
+                ]
+                [ text "Undo" ]
+            , button
+                [ classes [ btn, bold, (Classes.disabled $ not isEnabled) ]
+                , onClick $ const $ Just ResetSimulator
+                ]
+                [ text "Reset" ]
+            ]
+        ]
+  where
+  -- FIXME: I think the contract should always be valid if we are in the simulation
+  isEnabled = isContractValid state
+
+  possibleActions = view (_marloweState <<< _Head <<< _executionState <<< _SimulationRunning <<< _possibleActions <<< _Newtype) state
+
+  kvs :: forall k v. Map k v -> Array (Tuple k v)
+  kvs = Map.toUnfoldable
+
+  vs :: forall k v. Map k v -> Array v
+  vs m = map snd (kvs m)
+
+  lastKey :: Maybe Party
+  lastKey = map (\x -> x.key) (Map.findMax possibleActions)
+
+  sortParties :: forall v. Array (Tuple Party v) -> Array (Tuple Party v)
+  sortParties = sortWith (\(Tuple party _) -> party == otherActionsParty)
+
+  actionsForParties :: Map Party (Map ActionInputId ActionInput) -> Array (HTML p Action)
+  actionsForParties m = map (\(Tuple k v) -> participant state isEnabled k (vs v)) (sortParties (kvs m))
+
+participant ::
+  forall p.
+  State ->
+  Boolean ->
+  Party ->
+  Array ActionInput ->
+  HTML p Action
+participant state isEnabled party actionInputs =
+  li [ classes [ ClassName "participant-a", noMargins ] ]
+    ( [ h6_ [ em_ title ] ]
+        <> (map (inputItem state isEnabled partyName) actionInputs)
+    )
+  where
+  title =
+    if party == otherActionsParty then
+      -- QUESTION: if we only have "move to slot", could we rename this to "Slot Actions"?
```

Comment:
I am neutral on this, so for me leave it as it is

Reply:
good :)